packages/router/src/lib/operations/auction.ts

```diff
@@ -163,20 +163,45 @@ export const newAuction = async (
   // getting the swap rate from the receiver side config
   const amountReceived = await getReceiverAmount(amount, inputDecimals, outputDecimals);
 
-  const balance = await contractReader.getAssetBalance(receivingAssetId, receivingChainId);
-  logger.info({ method, methodId, balance: balance.toString() }, "Got asset balance");
-  if (balance.lt(amountReceived)) {
+  let availableLiquidity = await contractReader.getAssetBalance(receivingAssetId, receivingChainId);
+  logger.info(
+    { method, methodId, requestContext, balance: availableLiquidity.toString() },
+    "Got available liquidity from contract reader",
+  );
+
+  // get cached outstanding amount
+  const outstandingLiquidity = await cache.getOutstandingLiquidity({
+    assetId: receivingAssetId,
+    chainId: receivingChainId,
+  });
+  logger.info(
+    { method, methodId, requestContext, balance: outstandingLiquidity.toString() },
+    "Got available liquidity from contract reader",
+  );
+
+  availableLiquidity = availableLiquidity.sub(outstandingLiquidity);
+
+  if (availableLiquidity.lt(amountReceived)) {
```

Comment:
We'll need to configure this properly. It won't work with lifi nxtp as-is, and will quickly cause the router to erroneously report that there's not enough liquidity. load tests will not catch this because they request a quote 1x per transfer

Reply:
Yes, we need to change the LiFi implementation to match this, but it seems like this is how we need the functionality to be implemented on our end.

Reply:
What if it's a dry run?

Reply:
Dry run will not remove outstanding liquidity. That was the idea behind dry run to begin with. I think this is a bug, thanks for pointing out.